.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,40 @@
+name: CodeQL
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  schedule:
+    - cron: 00 00 * * 00
```

Comment:
Is the schedule needed? It seems to me that if we run the check on push to main that'd be enough, unless there's some value to run it periodically?

Reply:
Required if running on a schedukle like we do in the Terraform Provider repository.

<img width="1487" alt="image" src="https://github.com/user-attachments/assets/d240374b-51e3-455d-ac2f-1af86ac816b3">